### DISCUSSION TITLE
Ability to use annotations for safelist and other directives

### DISCUSSION THREAD

**Discussion by inilim** [tailwindlabs/tailwindcss#12454] [Ideas]
Example:
```
/**
 * @tw-safe m-[1px] alert-success
 */
const userInput = 'success';
const alertClass = `alert-${userInput}`;
```

```
<?php
/**
 * @tw-safe m-[1px] alert-success
 */
$type = 'success';
$class_name = 'alert-' . $type;
```

**Comment by wongjn** [comment]
As long as this file is covered by the `content` file globs, this should already work. This is because Tailwind scans files as plain text files (as if the file ended with `.txt` or similar) and thus should pick up the classes within the comments.
